### DISCUSSION TITLE
Is it possible to conditionally hide Launch button.

### DISCUSSION THREAD

**Discussion by lukeskrz** [wixtoolset/issues#6578] [Questions] [answered]
Hello,

I'm using v3.11.2 of the toolset, and trying to figure out if it is possible to conditionally hide the "Launch" button on the success page of the `WixStandardBootstrapperApplication` using `RtfLargeLicense`?
Looking through the [source](https://github.com/wixtoolset/wix3/blob/376423b8101f4b59ee865e8a255cfe190fa5a7f1/src/ext/BalExtension/wixstdba/WixStandardBootstrapperApplication.cpp#L2535) it looks like the button is enabled/visible if the `LaunchTarget` variable exists and contains at least one character.
I had considered defining the `LaunchTarget` variable in my bundle with a default path and making it overridable
`<Variable Name="LaunchTarget" Value="[InstallLocation]\$(var.ProductName)\$(var.ExeName)" bal:Overridable="yes" />`
with the idea that I could pass `LaunchTarget=` on the command line to turn it off.

I see in the logs the `LaunchTarget` variable does get set to `''` when I pass in the command line, but the button then still appears.

Is there anyway this could be configured to work?

**Comment by rseanhall** [comment]
I expected your approach to work. You didn't provide a fully reproducible example so the error is probably in something you haven't explained here.

Is overriding the variable on the command line good enough or were you looking for something different? A new feature probably wouldn't be added to v3.x but it could be added for v4.

**Comment by lukeskrz** [comment, marked as answer]
Thanks for the response.
Here's a simple example bootstrapper and dummy MSI. By default if you click the `Launch` button it executes notepad. 
I can call it like `BootstrapperTest LaunchTarget=cmd.exe` to have it launch the cmd window instead.
However, if I run `BootstrapperTest LaunchTarget=` the `Launch` button is still visible.

[BootstrapperTest.zip](https://github.com/wixtoolset/issues/files/7164927/BootstrapperTest.zip)

Overriding the command line would probably be enough, although I have found a workaround with having the button launch a small helper process that launches my application in the correct manner.

**Reply by rseanhall** [reply]
Looks like the comment in `BalStringVariableExists` is wrong, a string variable with a value of an empty string will still return `E_MOREDATA` because `cch` includes 1 for the null terminator character. You'll have to write a [BAFunctions](https://github.com/wixtoolset/wix3/tree/376423b8101f4b59ee865e8a255cfe190fa5a7f1/src/burn/Samples/bafunctions) dll in order to undefine the variable (set the string value to NULL) in v3.

**Reply by lukeskrz** [reply]
Ah ok. Thanks for your help.
